service/src/main/java/com/ge/predix/acs/privilege/management/dao/AttributeTraversalException.java

```diff
@@ -0,0 +1,14 @@
+package com.ge.predix.acs.privilege.management.dao;
+
+public class AttributeTraversalException extends RuntimeException {
```

Comment:
After reading the code a bit more, it seems to me that the exception occurs when a traversal limit is reached. So why not call it something that reflects why it actually occurred. Such as "TraversalLimitException" or similar? What does attribute resolution have to do with it? If I was reading this code and saw that the exception incorporated "limit" or "LimitExceeded" I would have a better understanding of what the exception is for. In other words, name the exception after an event not an operation.